src/app/private/scripts/factories/fileUploadFactory.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+angular.module('playerApp')
+  .factory('fileUpload', ['$timeout', 'toasterService', function ($timeout, toasterService) {
+    return {
+      /**
+       * @method initializeFineUploader
+       * @desc initialize FineUploader third party plugin
+       * @memberOf Factory.fineUploader
+       * @param {Object}  option - Option object to invoke controller callback function
+       * @returns {Callback} Trigger callback function
+       */
+      createFineUploadInstance: function (options) {
```

Comment:
having the uploader type name in the method does not prevent strong coupling. having the options sent from the controller doesn't prevent abstraction leak.

the factory method should be called when the app is initialised. app-level config options should be passed in and an instance created. this instance is available as a partially configured factory. this factory can then be called with controller specific options to get the uploader instance.